### PR TITLE
Behat use project api prefix

### PR DESCRIPTION
**Pull request by Prometee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | none
| License         | MIT

Give the ability to use Behat API tests with a different api prefix, useful if you don't use the default prefix `/api/v2` or simply wanted to use some Behat services with another APIP installation.

### CODE REVIEW

**Comment by Prometee** [comment]
@Ferror yes, I was trying to solve some of the new issues that this PR is spotlighting.

There was some `'api/v2/....'` uri set before instead of `'/api/v2'` making some errors not showing up before.
Some 404 responses are now showing the right error expected message and a 500 response code.

I will try to fix them as well as trying to fit to the initial swagger API doc definition like this one :
image
We can see here that a 404 is expected instead of a 500 with the error message : `"No user found with reset token: itotallyforgotmypassword"`

**Comment by Prometee** [comment]
@Ferror I'm rebasing my PR with the new `RequestFactory` (great addition ^^), I think the `$apiRoute` can be now centralized there. What do you think about that ?

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:41]:
```suggestion
        private ApiClientInterface $shopClient,
        private ApiClientInterface $adminClient,
```

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:44]:
```suggestion
        private SharedStorageInterface $sharedStorage,
        private IriConverterInterface $iriConverter,
```

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:47]:
```suggestion
        private string $apiUrlPrefix,
```

File: src/Sylius/Behat/Client/ApiPlatformSecurityClient.php
Comment (Ferror) [inline src/Sylius/Behat/Client/ApiPlatformSecurityClient.php:28]:
```suggestion
        private AbstractBrowser $client,
        private SharedStorageInterface $sharedStorage,
        private string $apiUrlPrefix,
        private string $section,
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php:40]:
```suggestion
        private string $apiUrlPrefix,
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php:39]:
```suggestion
        private string $apiUrlPrefix,
```

File: src/Sylius/Behat/Context/Api/Shop/ChannelContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/ChannelContext.php:30]:
```suggestion
        private string $apiUrlPrefix,
```

File: src/Sylius/Behat/Context/Api/Shop/HomepageContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/HomepageContext.php:27]:
```suggestion
        private string $apiUrlPrefix,
```

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php
Comment (Ferror) [inline src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: Do you mind creating additional PR with that change? I think we might want to remove occurrence of 500 in our API 😜

**Review by Ferror — COMMENTED** [review]
I got this small code-style requests and we gonna merge it <3

**Comment by Ferror** [comment]
> I think, a better work is needed to centralize the url prefix usage to avoid adding it to all of those Behat contexts. WDYT ?

Yea I was thinking about some sort UriFactory for all these sprintf's functions as they are all similar and now require dependency :)

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php
Comment (Prometee) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: Yes, I thought about replacing the Assert throwing the 500 by a NotFoundException. Is that the way to do it ?

Comment (Ferror) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: Yea, that's it :)

Comment (Zales0123) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: Definitely a separate PR would be welcomed, let's focus on api prefix in this one 🖖

**Review by Zales0123 — COMMENTED** [review]
A few more simple changes and it's ready to go 👍 thanks, @Prometee!

**Comment by Prometee** [comment]
@Zales0123  @Ferror Waiting for the build to finish and it can be merged.

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php
Comment (Prometee) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: @Zales0123 @Ferror #13902 created to fix it.

**Review by Ferror — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php
Comment (lchrusciel) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: Let's merge #13902 at a separate pace. May I have a silly question? Why did you change it? I mean, I understand that 404 will be returned from this endpoint only after #13902 will be merged, but yet the current master build is green without this or #13902 change. Why?

Comment (Prometee) [reply src/Sylius/Behat/Context/Api/Shop/LoginContext.php:275]: @lchrusciel the response was a 404 because the route was not found :

```json
{
    "code":404,
    "message":"No route found for \u0022PATCH http:\/\/localhost\/api\/v2\/shop\/reset-password-requests\/api\/v2\/shop\/reset-password-requests\/itotallyforgotmypassword\u0022 (from \u0022http:\/\/localhost\/api\/v2\/shop\/reset-password-requests\/itotallyforgotmypassword\u0022)"
}
```

My changes is altering the route from `api/v2/...` to `/api/v2/...` that's why the real error is showing.

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, Francis! :tada: We could also thing about backporting it to 1.10 (before latest changes in Behat api clients) :rocket:

**Pull request merged**
